solidity/contracts/bridge/Deposit.sol

```diff
@@ -272,6 +367,40 @@ library Deposit {
         );
     }
 
+    /// @notice Sibling of the `revealDeposit` function. This function allows
+    ///         to reveal a P2(W)SH Bitcoin deposit with 32-byte extra data
+    ///         embedded in the deposit script. The extra data allows to
+    ///         attach additional context to the deposit. For example,
+    ///         it allows a third-party smart contract to reveal the
+    ///         deposit on behalf of the original depositor and provide
+    ///         additional services once the deposit is handled. In this
+    ///         case, the address of the original depositor can be encoded
+    ///         as extra data.
+    /// @param fundingTx Bitcoin funding transaction data, see `BitcoinTx.Info`.
+    /// @param reveal Deposit reveal data, see `RevealInfo struct.
+    /// @param extraData 32-byte deposit extra data.
+    /// @dev Requirements:
+    ///      - All requirements from `revealDeposit` function must be met,
+    ///      - `extraData` must not be bytes32(0),
+    ///      - `extraData` must be the actual extra data used in the P2(W)SH
+    ///        BTC deposit transaction.
+    ///
+    ///      If any of these requirements is not met, the wallet _must_ refuse
+    ///      to sweep the deposit and the depositor has to wait until the
+    ///      deposit script unlocks to receive their BTC back.
+    function revealDepositWithExtraData(
```

Comment:
It's not only about the vault. Most importantly, the `depositor` embedded in the script is the receiver of minted TBTC (or `Bank`'s balance). If something has to happen with minted TBTC automatically, the `depositor` must be aware of the deposit. In most cases, the `depositor` will also need to record some information upon deposit reveal (e.g. current fee values) to properly do the automation on minted TBTC (e.g. compute the net amount of the deposit). Requiring that only the `depositor` can reveal guarantees that the deposit always passes through the `depositor` before reaching the `Bridge`.

Let's use Base as an example: While making a deposit, we are setting the `BaseDepositor` contract as `depositor` and the Base EOA as the original depositor in extra data. If we allow anyone to reveal, a third party can just reveal that deposit to the `Bridge` and bypass the `BaseDepositor` contract. The `BaseDepositor` will receive minted TBTC but will not be aware of the deposit from the very beginning. Although it's probably possible to deliver that information later, this can be very tricky in some cases and complicates the relayer bot. Moreover, `BaseDepositor` hasn't had a chance to capture the `Bridge` state upon deposit reveal so doing the right thing with minted TBTC can be hard. This example also shows why we need the depositor contract for Base: it's not only about revealing the deposit but also about moving back minted TBTC back to Base.

Last but not least, I'd say that not having the vault in the script is more flexible and safe. What if the vault is set as untrusted due to a critical vulnerability? If the vault is embedded in the script, the deposit can be revealed there at any time by anyone, far before the refund. The vault cannot be changed in that case. If we don't embed, the deposit can always be revealed with vault `address(0)`. 

That said, I see more problems than advantages with letting anyone reveal the deposit. The `revealDepositWithExtraData` covers the flexibility gap that was caused by this requirement in the original `revealDeposit` and still provides strong security guarantees.